18_regexp_lab/02.py

```diff
@@ -1,6 +1,29 @@
-"""
-Write a function toCamelCase(text) that returns the text camel cased
-Write a function to_underscore(text) that returns the text with 
-underscores between words
-"""
-
+
+import re
+
+def camelCase(text):
+    res = re.search(r'(\w+)_(\w+)', text)
+    if res == None: 
+        print text, "not have _"   
+    else:
+        upper = re.sub((r'_(\w)') ,
+        lambda m: m.group(0).upper()
+        , text)
+        camel_case_word = re.sub('_' , '', upper)
+        return camel_case_word
+
+print camelCase('hi_there_hoe_r_u')
+
+def underline(text1):
+    res1 = re.search(r'([a-z])([A-Z])', text1)
+    if res1 == None:
+        print text1, "not have big letter"
+    else:
+        lower = re.sub('([A-Z])',
+        lambda n: '_' + n.group(0).lower()
+        , text1) 
+        return lower
+
+
+
+print underline('hiThereHowAre')
```

Comment:
Nice.
One note: when you spread a function call on multiple lines, it is going to be more readable if the 2nd and further lines are indented. Even if that is not required by Python. Like this:

```
        upper = re.sub((r'_(\w)'),
            lambda m: m.group(0).upper(),
            text)
```

though I'd probably have written this in a single line. It is only 70 char wide.

Also there is no reason to call the internal variable in the second function 'res1'. That could be called 'res' too. Those two variables in the two functions are independent anyway.